### PR TITLE
Add filter setting for yuv image.

### PR DESCRIPTION
**Pull request by JerryShih**
r? @kvark @glennw @nical 

Some reftests in gecko need the nearest sampling. Bypass the filter setting for yuv image.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1310)

### CODE REVIEW

**Comment by kvark** [comment]
Is there ever a case where YUV would need linear sampling? I'd expect this to produce colouring artefacts.
Other than this question, the code looks good and ready to merge.

**Comment by JerryShih** [comment]
@kvark 
I hit an assertion with this branch now. So, please don't merge this patch.

**Comment by JerryShih** [comment]
@kvark 
> Is there ever a case where YUV would need linear sampling? I'd expect this to produce colouring artefacts.
> Other than this question, the code looks good and ready to merge.

The yuv image uses  "ImageRendering::Auto" now. That will use TextureFilter::Linear.
So, I add an additional option for the filter setting.

                let filter = match request.rendering {
                    ImageRendering::Pixelated => TextureFilter::Nearest,
                    ImageRendering::Auto | ImageRendering::CrispEdges => TextureFilter::Linear,
                };

**Comment by kvark** [comment]
@JerryShih I understand what you did. My question is - do we ever need linear sampling on YUV?

**Comment by JerryShih** [comment]
> @JerryShih I understand what you did. My question is - do we ever need linear sampling on YUV?

I don't know. But, maybe we will have the better quality when we enlarge the video size.

**Comment by JerryShih** [comment]
@kvark 
r? again for the https://github.com/servo/webrender/pull/1310/commits/7fa2494f13a2d5a0ea5fcf9b0670ec5a359e7ce2

**Comment by kvark** [comment]
I thought a bit about using linear sampling on YUV in general, and talked to Markus.
TL;DR: it should be supported, but we may need further enhancements in the shader to make it look correct.

@bors-servo r+
